batchflow/tests/config_basic_test.py

```diff
@@ -0,0 +1,349 @@
+"""
+Each function tests specific Config class method.
+"""
+
+import sys
+import pytest
+
+sys.path.append('../..')
+
+from batchflow import Config
+
+def test_dict_init():
+    """
+    Tests Config.__init__() using input of dictionary type.
+    For inner structure check Config.flatten() is used.
+    """
+
+    #Slashed-structured dictionary initialization
+    init_dict = {'a' : 1, 'b/c' : 2, 'b/d' : 3}
+    exp_flat = {'a': 1, 'b/c': 2, 'b/d': 3}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+    #Nested-structured dictionary initialization
+    init_dict = {'a' : {}, 'b' : {'c' : 2, 'd' : 3}}
+    exp_flat = {'a': {}, 'b/c': 2, 'b/d': 3}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+    #Mixed-structured dictionary initialization
+    init_dict = {'a' : None, 'b/c' : 2, 'b' : {'d' : 3}}
+    exp_flat = {'a': None, 'b/c': 2, 'b/d': 3}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+    #Config-structured dictionary initialization
+    init_dict = {'a' : Config({'b' : 2})}
+    exp_flat = {'a/b': 2}
+    config = Config(init_dict)
+    assert config.flatten() == exp_flat
+
+def test_dict_init_bad():
+    """
+    Tests Config.__init__() using BAD input of dictionary type.
+    """
+
+    #Int-keyed dictionary initialization
+    init_dict = {0 : 1}
+    with pytest.raises(Exception):
```

Comment:
It is a very broad exception to catch

Reply:
Corrected by adding the appropriate check in config.py.
Corresponding tests now expect _TypeError_ exception.